chains/evm/calls/events/listener.go

```diff
@@ -30,8 +35,14 @@ func NewListener(client ChainClient) *Listener {
 }
 
 func (l *Listener) FetchDeposits(ctx context.Context, contractAddress common.Address, startBlock *big.Int, endBlock *big.Int) ([]*Deposit, error) {
+	tp := otel.GetTracerProvider()
```

Comment:
I would probably prefer this being initialized here https://github.com/ChainSafe/chainbridge-core/blob/e55b5cec4f6f346341f4ec6b8dd2667eed2e0156/chains/evm/listener/event-handler.go#L40 and then context passed around the FetchDeposits and HandleDeposit functions. 
`FromContext` function should be used to reuse the span so we have one span for the whole deposit and the only initialized  one should be in the event-handler IMO and we pass everything by context.